linalg/hypre.cpp

```diff
@@ -2779,6 +2779,11 @@ void HypreBoomerAMG::SetDefaultOptions()
    HYPRE_BoomerAMGSetTol(amg_precond, 0.0);
 }
 
+void HypreBoomerAMG::SetMaxIter(int maxIter)
+{
+   HYPRE_BoomerAMGSetMaxIter(amg_precond, maxIter);
+}
+
```

Comment:
I think that one of the philosophies of MFEM is to do the iterative processes within MFEM constructs while preconditioners may come from outside. This can probably be useful if for example you want to precondition by 2 Vcycles. However, that can also be resolve within what MFEM already provides.

I'm actually negative towards this addition, because it would allow too much unregulated freedom.

Reply:
I reverted these changes.